Sources/Markdown/Base/Markup.swift

```diff
@@ -183,14 +183,16 @@ extension Markup {
 
     /// Returns the child at the given position if it is within the bounds of `children.indices`.
     ///
-    /// - Complexity: `O(childCount)`
+    /// - Complexity: `O(1)`
     public func child(at position: Int) -> Markup? {
         precondition(position >= 0, "Cannot retrieve a child at negative index: \(position)")
-        guard position <= raw.markup.childCount else {
+        guard position < raw.markup.childCount else {
             return nil
         }
-        var iterator = children.dropFirst(position).makeIterator()
-        return iterator.next()
+        let rawChild = raw.markup.child(at: position)
+        let absoluteRawMarkup = AbsoluteRawMarkup(markup: rawChild, metadata: raw.metadata.firstChild())
```

Comment:
This is not correct. You’re using the first child’s metadata for the nth child.

Reply:
Is this incorrect in the iterator then too or am I misunderstanding something here?

https://github.com/apple/swift-markdown/blob/2ba1ed5/Sources/Markdown/Base/MarkupChildren.swift#L23